### PR TITLE
Feature/researcher-session-outputs

### PR DESCRIPTION
**Pull request by JessUWE**
## Summary
This PR adds a complete researcher interface for managing ACRO outputs, allowing researchers to:
- Work with empty output folders (scaffolding support)
- Add comments and exception requests to outputs
- Save draft sessions and finalize for output checker review
- Upload, edit, and delete outputs
- View PDFs and document files inline

### Researcher Interface (`researcher_index.html`)
- Two-role landing page (Researcher vs Output Checker)
- Output list sidebar with upload zone
- Detail panel with comments, exceptions, and checklist
- Modal dialogs for output management and session finalization

### Frontend JavaScript (`researcher.js`)
- Session state management
- Comment and exception request handling
- Output CRUD operations with file uploads
- Pre-submission checklist validation
- Modal interaction and form handling

### Document File Support
- PDF viewing
- Word/Excel document
- File checksums and validation
- MIME type formatting for user display

### Test Coverage
- 15 new test files covering:
  - Document file handling
  - Session save/load/finalize operations
  - Output add/edit/delete flows
  - Error handling and edge cases
  - View branching and coverage

## Changes
- Add `sacro/views.py` researcher endpoints and helpers
- Add `assets/src/scripts/researcher.js` 
- Add `sacro/templates/researcher_index.html` and supporting templates
- Add document file utilities to `_utils.js` and `_file-elements.js`
- Update `sacro/models.py` to restore validation for non-empty results
- Update `sacro/urls.py` with new routes
- Update documentation and dependencies

### CODE REVIEW

File: data/XandY.jfif [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-  <path d="M30.85 112.83c.23.52 1.35 2.36 1.35 2.37-1.26 14.52-23.93 16.36-23.93 16.36h-.01c-.28-.37-1.14-1.44-1.66-2.2l-.47-.7C2.04 122.46 0 115.16 0 106.72h29.64c0 2.23.4 4.25 1.19 6.05l.02.06Z"/>
-</g>
-<g style="fill:#5971f2;">
-  <path d="M32.33 115.34c.13 3.76 52.33 21.82 53.17 21.86a55.2 55.2 0 0 1-9.12 5.69c-7.44 3.55-16.23 5.33-26.38 5.33-10.14 0-18.97-1.71-26.48-5.14-6.3-2.87-11.38-6.7-15.26-11.52 11.4-2.8 22.6-6.53 23.95-16.37l.12.15Z"/>
-</g>
-<g style="fill:#5324b3;">
-  <path d="M98.46 92.61a51.48 51.48 0 0 1 1.54 12.92c0 8.44-2.07 15.88-6.23 22.34a39.5 39.5 0 0 1-8.32 9.31 94.76 94.76 0 0 1-35.17-11.38c-2.26-1.25-4.42-2.7-6.54-4.06 2.09.43 4.36.79 6.85.79 6.2 0 11.04-1.38 14.53-4.15 3.49-2.77 5.24-6.59 5.24-11.46 0-3.56-.94-6.69-2.81-9.4 9.13-2.73 19.3-4.56 30.9-4.9l.01-.01Z"/>
-</g>
-<g style="fill:#ff369c;">
-  <path d="m2.13 45.6 29.43-25.33c.25-.23.02.01.02.01l10.39 6.3.04.01a15.12 15.12 0 0 0-5.84 3.05c-.37.32-1.07 1.05-1.07 1.05-2.32 2.58-3.48 6.12-3.48 10.61a15.1 15.1 0 0 0 3.85 10.18c2.57 2.97 6.3 4.98 11.17 6.03l14.43 3.36a67.72 67.72 0 0 1 17.44 6.42c-15.35.57-27.82-1.82-40.27-6.03C25.67 57.01 3.82 47.6 2.26 47.16l-.13-1.56Z"/>
```
Comment (rpreen) [inline data/XandY.jfif]: This file consists of only whitespace or line ending changes and should not be in this PR.

File: justfile [hunk 2]
```diff
@@ -121,6 +121,14 @@ test-e2e: devenv test-outputs collectstatic
...
     npm run cypress:open
 
+# run black formatter
+black: devenv
+    $BIN/black .
+
+# run ruff linter
+ruff: devenv
+    $BIN/ruff check --fix .
+
```
Comment (rpreen) [inline justfile]: Build/style changes should be in a separate PR.

File: requirements.dev.in [hunk 3]
```diff
@@ -5,7 +5,7 @@
 # pip-compile --generate-hashes --output-file=requirements.dev.txt requirements.dev.in
 
 acro==0.4.0
-black
+black==24.10.0
```
Comment (rpreen) [inline requirements.dev.in]: Dependency changes should be in a separate PR.

File: cypress/e2e/reject-all.cy.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 describe("Start to finish user journey for rejecting all outputs and downloading the empty zip", () => {
   beforeEach(() => {
-    cy.visit("http://localhost:8000");
+    cy.visit("http://localhost:8000/checker/?path=outputs/results.json");
```
Comment (rpreen) [inline cypress/e2e/reject-all.cy.js]: What is the reason that this has been changed to point to a specific json file?
